### DISCUSSION TITLE
TypeError: '>' not supported between instances of 'NoneType' and 'NoneType'

### DISCUSSION THREAD

**Discussion by deemok7** [talmolab/sleap#1332] [Help!] [answered]
Dear SLEAP developers,

I am writing to report an error that I encountered while using the SLEAP package. The error message is as follows:

Traceback (most recent call last):
File "C:\Python37\lib\runpy.py", line 193, in _run_module_as_main
"main", mod_spec)
File "C:\Python37\lib\runpy.py", line 85, in run_code
exec(code, run_globals)
File "D:\vEnv\sleapEnv\Scripts\sleap-train.exe_main.py", line 7, in <module>
File "D:\vEnv\sleapEnv\lib\site-packages\sleap\nn\training.py", line 1947, in main
trainer.train()
File "D:\vEnv\sleapEnv\lib\site-packages\sleap\nn\training.py", line 906, in train
self.setup()
File "D:\vEnv\sleapEnv\lib\site-packages\sleap\nn\training.py", line 892, in setup
self._setup_model()
File "D:\vEnv\sleapEnv\lib\site-packages\sleap\nn\training.py", line 723, in _setup_model
self.data_readers.training_labels_reader
File "D:\vEnv\sleapEnv\lib\site-packages\sleap\nn\data\pipelines.py", line 364, in make_base_pipeline
provider=data_provider,
File "D:\vEnv\sleapEnv\lib\site-packages\sleap\nn\data\resizing.py", line 319, in from_config
max_height, max_width = provider.max_height_and_width
File "D:\vEnv\sleapEnv\lib\site-packages\sleap\nn\data\providers.py", line 140, in max_height_and_width
return max(video.shape[1] for video in self.videos), max(
TypeError: '>' not supported between instances of 'NoneType' and 'NoneType'

I believe that this error is related to the resizing of the training data. Specifically, it seems that the package is unable to determine the maximum height and width of the video frames, which leads to a TypeError.

I would appreciate any guidance or advice on how to resolve this issue. Thank you for your attention to this matter.

Best regards,

**Comment by roomrys** [comment, marked as answer]
Hi @mmmahdiii,

This error occurs when the video is not loaded correctly in SLEAP. In one case #961, the user found that the video was corrupted and redownloading and importing the video helped resolve the issue. In another case #1259, one user experience this issues intermittently (which is the most difficult type of issue to diagnose) and another stated that moving the video to a different directory caused the issue to occur. In yet another case #1339, the user did not unzip the directory containing the videos, hence SLEAP was unable to load the videos correctly.

Collecting some more info on the problem, does your case fall into any of the three situations above (video corruption, intermittent error, or missing video)? Since the code that produces the error is done in a loop, it might be a bit difficult to determine which video is causing problems (unless you only have a few videos), but I'll run through how to do the diagnosis:

### Video corruption
Possible yes if SLEAP (or another video reader) has trouble reading frames from one of your videos?

### Intermittent Error
Yes if this error only occurs sometimes, but not others. A stochastic bug, which is more difficult to find, but still possible!

### Missing Video
SLEAP should pop up a dialog to find missing videos if the project is opened in the GUI. The user must then save the project to update the video paths. But, if the video is opened via the API, SLEAP will still load the project even if it cannot find the video at the path it was last saved. There is an option to load the labels file via [`Labels.load_file`](https://sleap.ai/api/sleap.io.dataset.html#sleap.io.dataset.Labels.load_file) and provide a `video_search` as a list of paths where the videos may be located at (the paths can point to directories or files).
```
from sleap import Labels


filename = ...
path_a = ...
path_b = ...
labels = Labels.load_file(filename, video_search=[path_a, path_b])
```

I would think the __Missing Video__ case is the most likely since you seem to be running a script for training. Let us know if any of the above help.

Thanks,
Liezl

**Reply by deemok7** [reply]
Sorry for the delayed response. I had to rename all of the video file names, so the video names in the SLP file were different from their corresponding video paths. Therefore, I had to change the video names in the SLP project.

**Reply by roomrys** [reply]
@mmmahdiii Ah ok, 

I thought your question in #1343 looked similar. I'll follow up there as well.

Liezl
